.github/workflows/deploy.yml

```diff
@@ -49,42 +50,56 @@ jobs:
           docker build \
             --build-arg APP_BUILD_DATE=${{ env.BUILD_DATE }} \
             --build-arg APP_BUILD_TAG=${{ steps.vars.outputs.build_tag }} \
-            --build-arg APP_GIT_COMMIT=${{ github.sha }} \
-            -t ${{ vars.ECR_URL }}:${{ github.sha }} .
+            --build-arg APP_GIT_COMMIT=$SHA \
+            -t ${{ vars.ECR_URL }}:$SHA .
 
       - name: Push to ECR
-        run: |
-          docker tag ${{ vars.ECR_URL }}:${{ github.sha }} ${{ vars.ECR_URL }}:staging.latest
-          docker tag ${{ vars.ECR_URL }}:${{ github.sha }} ${{ vars.ECR_URL }}:production.latest
-          docker push ${{ vars.ECR_URL }}:${{ github.sha }}
-          docker push ${{ vars.ECR_URL }}:staging.latest
-          docker push ${{ vars.ECR_URL }}:production.latest
+        run: docker tag ${{ vars.ECR_URL }}:$SHA
 
   deploy-staging:
     runs-on: ubuntu-latest
     needs: build
     environment: staging
 
+    permissions:
+      id-token: write # This is required for requesting the JWT
+      contents: read  # This is required for actions/checkout
+
     env:
       KUBE_NAMESPACE: ${{ secrets.KUBE_NAMESPACE }}
+      KUBE_CERT: ${{ secrets.KUBE_CERT }}
```

Comment:
For consistency you might want to move this back to the authenticate to cluster step as that's how it's being done for the production deploy.